General/util.py

```diff
@@ -0,0 +1,39 @@
+import copy
+
+import django.forms
+
+
+class SelectWithDisabled(django.forms.Select):
```

Comment:
That's probably nicer but I suspect that it will require more adjustments to the superclass because we also need to modify the validation that checks the choice, as now invalid choices are combined with valid choices.